docs/dev/experimental-mode.adoc

```diff
@@ -19,74 +19,104 @@ Experimental mode can be enabled by setting odo preferences as well.
 odo preference set experimental true
 ```
 #### Working with experimental mode:
-Consider that the support for `odo push` using `--devfile` flag is under development.
-If the experimental mode is enabled, the `odo help push` will show the `--devfile` flag in `odo help push`
-```sh
-$ odo push -h
-Push source code to a component.
-
-Usage:
-  odo push [component name] [flags]
-
-Examples:
-  # Push source code to the current component
-  odo push
-  
-  # Push data to the current component from the original source.
-  odo push
-  
-  # Push source code in ~/mycode to component called my-component
-  odo push my-component --context ~/mycode
-
-Flags:
-      --config               Use config flag to only apply config on to cluster
-      --context string       Use given context directory as a source for component settings
-      --devfile string       Path to a devfile.yaml (default "./devfile.yaml")              <---- seen in help
-  -f, --force-build          Use force-build flag to force building the component
-  -h, --help                 Help for push
-      --ignore stringSlice   Files or folders to be ignored via glob expressions.
-      --show-log             If enabled, logs will be shown when built
-      --source               Use source flag to only push latest source on to cluster
-
-Additional Flags:
-  -v, --v Level              Log level for V logs. Level varies from 0 to 9 (default 0).
-      --vmodule moduleSpec   Comma-separated list of pattern=N settings for file-filtered logging
+
+##### Listing devfile components:
+When the experimental mode is enabled, `odo catalog list components` will list the devfile components alongside the OpenShift components.
```

Comment:
We should mention that the OpenShift/s2i components only show up on OpenShift clusters, and only the devfile components will be listed on Kubernetes clusters.

Reply:
Maybe also highlight that `odo catalog list components --all` will show all devfile components between both devfile registries (even non-supported ones)